src/core/referencingfeaturelistmodel.h

```diff
@@ -0,0 +1,96 @@
+/***************************************************************************
+  referencingfeaturelistmodel.h - ReferencingFeatureListModel
+
+ ---------------------
+ begin                : 1.3.2019
+ copyright            : (C) 2019 by David Signer
+ email                : david@opengis.ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef REFERENCINGFEATURELISTMODEL_H
+#define REFERENCINGFEATURELISTMODEL_H
+
+#include <QAbstractItemModel>
+#include <QPair>
+#include "qgsvectorlayer.h"
+#include "attributeformmodel.h"
+
+class QgsVectorLayer;
+class AttributeFormModel;
+
+class ReferencingFeatureListModel : public QStandardItemModel
+{
+  Q_OBJECT
+
+  /**
+   * The relation
+   */
+  Q_PROPERTY( QgsFeature feature WRITE setFeature READ feature NOTIFY featureChanged )
+  Q_PROPERTY( QgsRelation relation WRITE setRelation READ relation NOTIFY relationChanged )
+  Q_PROPERTY( QgsRelation associatedRelation WRITE setAssociatedRelation READ associatedRelation NOTIFY associatedRelationChanged )
+
+public:
+  explicit ReferencingFeatureListModel(QObject *parent = nullptr);
+
+  enum ReferencedFeatureListRoles
+  {
+    DisplayString = Qt::UserRole,
+    ReferencingFeature,
+    AssociatedReferencedFeature
+  };
+
+  QHash<int, QByteArray> roleNames() const override;
+  QModelIndex index(int row, int column, const QModelIndex &parent = QModelIndex()) const override;
+  QModelIndex parent(const QModelIndex &index) const override;
+  int rowCount(const QModelIndex &parent = QModelIndex()) const override;
+  int columnCount(const QModelIndex &parent = QModelIndex()) const override;
+
+  QVariant data(const QModelIndex &index, int role = Qt::DisplayRole) const override;
+
+  void setFeature( const QgsFeature &feature );
+  QgsFeature feature() const;
+
+  void setRelation( const QgsRelation &relation );
+  QgsRelation relation() const;
+
+  void setAssociatedRelation( const QgsRelation &relation );
```

Comment:
What's this? The n:m relation? In this case I would propose to call this setNmRelation to keep it consistent with QGIS

Reply:
On "Nm-Relation" I think about the network-relation containing the table in between - means a combination of the two relations. But this is from the "view" of the current feature the associated relation, so I called it like this. But no strong opinion - I can rename it without discussion if you like :-)